core/forkchoice.go

```diff
@@ -106,3 +113,15 @@ func (f *ForkChoice) ReorgNeeded(current *types.Header, header *types.Header) (b
 	}
 	return reorg, nil
 }
+
+// reorgNeededWithFastFinality compares justified block numbers firstly, backoff to compare tds when equal
+func (f *ForkChoice) reorgNeededWithFastFinality(current *types.Header, header *types.Header) (bool, error) {
+	_, ok := f.chain.Engine().(consensus.PoSA)
```

Comment:
if not ok, let us return `f.ReorgNeeded(current, header)` first, since it may not able to `GetJustifiedNumber`.

Reply:
accepted